src/main/java/me/kavin/piped/utils/ResponseHelper.java

```diff
@@ -625,6 +625,44 @@ public static byte[] commentsPageResponse(String videoId, String prevpageStr) th
 
     private static final Argon2PasswordEncoder argon2PasswordEncoder = new Argon2PasswordEncoder();
 
+    public static byte[] deleteUserResponse(String session, String pass) throws IOException {
+
+        if (StringUtils.isBlank(pass))
+            return Constants.mapper.writeValueAsBytes(new InvalidRequestResponse());
+
+        try (Session s = DatabaseSessionFactory.createSession()) {
+            User user = DatabaseHelper.getUserFromSession(session);
+            
+            if (user != null) {
+                String hash = user.getPassword();
+                boolean passMatch =
+                    (hash.startsWith("$argon2") && argon2PasswordEncoder.matches(pass, hash)) || bcryptPasswordEncoder.matches(pass, hash);
+
+                if (!passMatch)
+                    return Constants.mapper.writeValueAsBytes(new IncorrectCredentialsResponse());
+                
+                try {
+                    s.getTransaction().begin();
+
+                    s.createNativeQuery("delete from users_subscribed where subscriber = :id")
+                            .setParameter("id", user.getId()).executeUpdate();
+                    s.createNativeQuery("delete from playlists where owner = :ownerId")
+                            .setParameter("ownerId", user.getId()).executeUpdate();

```

Comment:
This doesn't delete playlists video contents however if I'm not wrong, ie. the `playlist_videos` table.

Reply:
You're right, I'll fix that.

Reply:
@FireMasterK it should be fixed now. I've manually verified that all DB tables are emptied on user deletion.

I've also improved `unsubscribeResponse()` a bit by avoding raw SQL statements. :slightly_smiling_face:

Reply:
> This doesn't delete playlists video contents however if I'm not wrong, ie. the playlist_videos table.

Actually, now that I think about it, is that a good idea? The videos in `playlist_videos` might be part of other playlists, which is why they're not deleted on normal playlist deletion I think.

Reply:
I would rather create a new method that's called when deleting a user. That new method would purge the `playlist_videos` table from videos that are not part of any playlist. It should also be called from `deletePlaylistResponse()`.

Reply:
Btw, I did the above and verified that it works.